enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -446,7 +447,13 @@ def signal_kernel(self, signum):
         else:
             raise RuntimeError("Cannot signal kernel. No kernel is running!")
 
-    def cleanup(self, connection_file=True):
+    def cleanup(self, restart=False):
+        """Clean up resources when the kernel is shut down"""
+        warnings.warn("Method cleanup(connection_file=True) is deprecated, use cleanup_resources(restart=False).",
```

Comment:
I don't think this is going to work without also modifying the jupyter_client dependency otherwise the call to `super().cleanup_resources()` will produce _AttributeError: 'super' object has no attribute 'cleanup_resources'_.  If we wanted to support older jupyter_client's we'd need to "plumb" `cleanup_resources()` as its own method chain, leaving `cleanup()` as-is.

I think it would be best to simply add a dependency on jupyter_client == 6.1.5 and then this looks good.

Reply:
Oh shoot. yes you are correct. i completely forgot about client version. nice catch.


This is how i am thinking
```
    def cleanup(self, restart=False):
        """Clean up resources when the kernel is shut down"""
        warnings.warn("Method cleanup(connection_file=True) is deprecated, use cleanup_resources(restart=False).",
                      FutureWarning)
        self.cleanup_resources(restart)

    def cleanup_resources(self, restart=False):
        """Clean up resources when the kernel is shut down"""

        # Note we must use `process_proxy` here rather than `kernel`, although they're the same value.
        # The reason is because if the kernel shutdown sequence has triggered its "forced kill" logic
        # then that method (jupyter_client/manager.py/_kill_kernel()) will set `self.kernel` to None,
        # which then prevents process proxy cleanup.
        if self.process_proxy:
            self.process_proxy.cleanup()
            self.process_proxy = None

        import jupyter_client as jc
        from distutils.version import LooseVersion
        if LooseVersion(jc.__version__) < LooseVersion('6.2'):
            return super(RemoteKernelManager, self).cleanup(connection_file=not restart)
        else:
            return super(RemoteKernelManager, self).cleanup_resources(restart)
```

Reply:
I think you came up with a solution along lines of my "method chain" comment - although I'd prefer to leave `cleanup()` as-is (the superclass call will produce the deprecation warning, so no need here).

Could you please add comments to each of the methods along the lines of...

for cleanup(): # This method has been deprecated in jupyter_client 6.1.5 and remains here for pre-6.2.0 jupyter_client installations.

for cleanup_resources(): # This method was introduced in jupyter_client 6.1.5 and will not be called until jupyter_client 6.2.0 has been released.

I just want to be sure it's clear to readers/maintainers these methods are mutually exclusive of each other.

Reply:
Done reset cleanup method to original one and added comments